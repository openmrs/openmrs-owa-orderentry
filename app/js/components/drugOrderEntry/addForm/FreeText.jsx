import React from 'react';
import PropTypes from 'prop-types';

const FreeText = ({
  fieldErrors,
  fields,
  options,
  handleChange,
  handleSubmit,
  handleCancel,
  handleValidation,
  activateSaveButton,
}) => {
  const { drugInstructions, dispensingQuantity, dispensingUnit } = fields;
  return (
    <div>
      <form className="simple-form-ui">
        <div className="flex-row">
          <div>
            <label name="notes">Complete Instructions</label>
            <textarea
              rows="4"
              cols="80"
              id="notes"
              name="drugInstructions"
              value={drugInstructions}
              onChange={handleChange}
            />
          </div>
        </div>
        <div className="flex-row">
          <div>
            <p className="left label-margin">
              <label>Dispense:</label>
            </p>
            <p className="left p-margin">
              <input
                className={`medium-input ${fieldErrors.dispensingQuantity && "illegalValue"}`}
                placeholder="Quantity"
                name="dispensingQuantity"
                id="dispenseQuantity"
                type="number"
                min="0"
                onBlur={handleValidation}
                onChange={handleChange}
                value={fields.dispensingQuantity} />
              {
                fieldErrors.dispensingQuantity ?
                  <span className="field-error">Required</span>
                  : ""
              }
            </p>
            <p className="left">
              <input
                className={(fields.dispensingQuantity && !fields.dispensingUnit) ? "illegalValue" : ""}
                placeholder="Units"
                name="dispensingUnit"
                id="drugDispensingUnits"
                list="dispensingUnits"
                size="20"
                value={fields.dispensingUnit}
                onBlur={handleValidation}
                onChange={handleChange} />
              <datalist id="dispensingUnits" >
                {
                  options.dispensingUnit &&
                    options.dispensingUnit.map(unit =>
                      (
                        <option key={unit.uuid} value={unit.display} />
                      ))
                }
              </datalist>
              {
                (fields.dispensingQuantity && !fields.dispensingUnit) &&
                  <span className="field-error">Required</span>
              }
            </p>
          </div>
        </div>
        <div className="flex-row">
          <div>
            <button
              type="button"
              className="cancel"
              onClick={handleCancel}
            >
            Cancel
            </button>
          </div>
          <div className="column-1">
            <button
              type="button"
              className="confirm"
              onClick={handleSubmit}
              disabled={activateSaveButton()}
            >
            Add
            </button>
          </div>
        </div>
      </form>
    </div>
  );
};

FreeText.propTypes = {
  fieldErrors: PropTypes.object.isRequired,
  handleValidation: PropTypes.func.isRequired,
  activateSaveButton: PropTypes.func.isRequired,
  fields: PropTypes.object.isRequired,
  options: PropTypes.object,
  handleCancel: PropTypes.func.isRequired,
  handleChange: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
};

FreeText.defaultProps = {
  options: {},
};

export default FreeText;
