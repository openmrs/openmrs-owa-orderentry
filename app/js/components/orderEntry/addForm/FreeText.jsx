import React from 'react';
import PropTypes from 'prop-types';

const FreeText = ({
  fields,
  allConfigurations,
  handleChange,
  handleSubmit,
}) => {
  const { completeInstructions, dispensingQuantity, dispensingUnit } = fields;
  return (
    <div>
      <form>
        <div>
          <div>
            <label name="instructions">Complete Instructions</label>
            <textarea
              id="instructions"
              name="completeInstructions"
              value={completeInstructions}
              onChange={handleChange}
              required
            />
          </div>
          <div>
            <div>
              <label>Dispense:</label>
            </div>
            <div>
              <label name="Quantity">Dose</label>
              <input
                type="number"
                id="Quantity"
                name="dispensingQuantity"
                min="0"
                value={dispensingQuantity}
                onChange={handleChange}
                required
              />
            </div>
            <div>
              <label name="dispenseUnits">Units</label>
              <input
                type="search"
                id="dispenseUnits"
                name="dispensingUnit"
                list="units"
                size="10"
                value={dispensingUnit}
                onChange={handleChange}
                required />
              <datalist id="units" >
                {
                  allConfigurations &&
                  (
                    allConfigurations.drugDispensingUnits &&
                  allConfigurations.drugDispensingUnits.map(unit =>
                    (
                      <option key={unit.uuid} value={unit.display} />
                    ))
                  )
                }
              </datalist>
            </div>
          </div>
          <div>
            <input type="button" className="button cancel" value="Cancel" />
            <input type="button" onClick={handleSubmit} className="button confirm right" value="Save" />
          </div>
        </div>
      </form>
    </div>
  );
};

FreeText.propTypes = {
  fields: PropTypes.object.isRequired,
  allConfigurations: PropTypes.shape({
    drugDispensingUnits: PropTypes.array,
  }),
  handleChange: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
};

FreeText.defaultProps = {
  allConfigurations: {},
};

export default FreeText;
