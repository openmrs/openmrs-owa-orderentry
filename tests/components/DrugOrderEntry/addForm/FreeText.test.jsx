import React from 'react';

import FreeText from '../../../../app/js/components/drugOrderEntry/addForm/FreeText';

const props = {
  activateSaveButton: jest.fn(),
  fields: {},
  fieldErrors: {},
  careSetting: {display: 'Outpatient'},
  options: {
    dispensingUnit: [{
      display: 'kits',
      uuid: 'ABC-56Y',
    }]
  },
  handleValidation: jest.fn(),
  handleChange: jest.fn(),
  handleCancel: jest.fn(),
  handleSubmit: jest.fn(),
};

describe('Test for Free text form', () => {
  it('should render component', () => {
    const wrapper = shallow(<FreeText {...props} />  );
    expect(wrapper).toMatchSnapshot()
  });
  it('should render options if they exist in state', () => {
    const wrapper = shallow(<FreeText {...props} />  );
    expect(wrapper.find('option')).toHaveLength(1);
  });
});
