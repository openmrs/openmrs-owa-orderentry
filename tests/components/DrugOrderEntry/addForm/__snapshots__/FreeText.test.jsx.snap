// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Free text form should render component 1`] = `
<div>
  <form
    className="simple-form-ui"
  >
    <div
      className="flex-row"
    >
      <div>
        <label
          name="notes"
        >
          Complete Instructions
        </label>
        <textarea
          cols="80"
          id="notes"
          name="drugInstructions"
          onChange={[MockFunction]}
          rows="4"
        />
      </div>
    </div>
    <div
      className="flex-row"
    >
      <div>
        <p
          className="left label-margin"
        >
          <label>
            Dispense:
          </label>
        </p>
        <p
          className="left p-margin"
        >
          <input
            className="medium-input undefined"
            id="dispenseQuantity"
            min="0"
            name="dispensingQuantity"
            onBlur={[MockFunction]}
            onChange={[MockFunction]}
            placeholder="Quantity"
            type="number"
          />
        </p>
        <p
          className="left"
        >
          <input
            className=""
            id="drugDispensingUnits"
            list="dispensingUnits"
            name="dispensingUnit"
            onBlur={[MockFunction]}
            onChange={[MockFunction]}
            placeholder="Units"
            size="20"
          />
          <datalist
            id="dispensingUnits"
          >
            <option
              key="ABC-56Y"
              value="kits"
            />
          </datalist>
        </p>
      </div>
    </div>
    <div
      className="flex-row"
    >
      <div>
        <button
          className="cancel"
          onClick={[MockFunction]}
          type="button"
        >
          Cancel
        </button>
      </div>
      <div
        className="column-1 pull-right-8"
      >
        <button
          className="confirm"
          onClick={[MockFunction]}
          type="button"
        >
          Add
        </button>
      </div>
    </div>
  </form>
</div>
`;
